src/main/java/com/google/sps/servlets/getData.java

```diff
@@ -55,133 +45,91 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 // #######################
 // # Retrieving the Data #
 // #######################
+
+   // debug long startingTime = System.currentTimeMillis();  avg time 1.35 seconds first attempt, .5 seconds for second attempt
       
    // Create a Java URL to connect to the CSV we want to read from.
-      URL url = new URL("https://raw.githubusercontent.com/nytimes/covid-19-data/master/live/us-counties.csv");
-      
-   // From that URL, send a "Get" request to retrieve the information.
-      URLConnection data = url.openConnection();
+      URL url = new URL("https://raw.githubusercontent.com/nytimes/covid-19-data/master/rolling-averages/us-counties-recent.csv");
       
-   // Create a Scanner that can read from our URL response.
-      Scanner input = new Scanner(data.getInputStream());
+   // Create a BufferedReader that can read from our URL. (faster)
+      BufferedReader input = new BufferedReader(new InputStreamReader(url.openStream()));
       
       
 // #######################   Notes:
-// # Processing the Data #      Since I want the data to be indexed by State AND County, I must use "Pairs."
-// #######################      Also, since the data we have for every county is also a pair, I'll use one in our data aswell.
-
-   // Create a hashmap to store our data, indexed by a pair of strings, that will store a pair of ints. (State, County) => (cases, deaths)
-      HashMap<Pair<String, String>, Pair<Integer, Integer>> dataMap = new HashMap<Pair<String, String>, Pair<Integer, Integer>>();
+// # Processing the Data #      We're dealing with rolling averages data now. File's a lot bigger, I'm not loading it all into memory.
+// #######################      I'm going to parse through the file, and based on user input, retrieve the necessary line from file.
       
    // Skip the first line of our file, it only contains table headers
```

Comment:
Here's where the biggest change happens. We went from reading a pretty small file, to a much bigger files (5 MB). Before, reading the whole file and using a hashmap wasn't such a bad idea, but now parsing all that extra and saving it wouldn't be all that useful- and I realized another solution I could use that would be way simpler anyway.

Instead, everything I'd do would now be through the file. 

First thing I had to do was skip a lot of old data- the file has all the data from last month onward, so I have to skip to the most recent data (yesterday). 

Then, once I'm there, instead of scanning through our data and making a hashmap, I'm instead just going to scan through line by line, and if I find a line that has the state and county we're looking for, we're good! This also means that we scan through less to find our result. 

Some of the code I think could be optimized more now that it's in a state that it works, but at the same time I think it's clean enough given its purpose. I might take a look at improving it later.